includes/class-sensei-modules.php

```diff
@@ -108,6 +108,94 @@ public function __construct( $file ) {
 		// Add custom navigation.
 		add_action( 'in_admin_header', [ $this, 'add_custom_navigation' ] );
 		add_filter( 'submenu_file', [ $this, 'highlight_menu_item' ] );
+
+		// Update module teacher meta when added to course.
+		add_action( 'added_term_relationship', [ $this, 'add_teacher_id_in_module_meta_when_added_to_course' ], 10, 3 );
+
+		// Remove module teacher meta when removed from course.
+		add_action( 'delete_term_relationships', [ $this, 'remove_teacher_id_from_module_meta_when_removed_from_course' ], 10, 3 );
+
+		// Update module teacher meta on course teacher update.
+		add_action( 'post_updated', [ $this, 'update_module_teacher_id_meta_on_post_teacher_update' ], 10, 3 );
+	}
+
+	/**
+	 * Add teacher id as term meta when a module is added to a course.
+	 *
+	 * @since $$next-version$$
+	 * @access private
+	 *
+	 * @param int     $post_ID      Post ID.
+	 * @param WP_Post $post_after   Post object following the update.
+	 * @param WP_Post $post_before  Post object before the update.
+	 */
+	public function update_module_teacher_id_meta_on_post_teacher_update( int $post_ID, WP_Post $post_after, WP_Post $post_before ) {
+		if ( 'course' !== get_post( $post_ID )->post_type ) {
+			return;
+		}
+
+		if ( $post_after->post_author !== $post_before->post_author ) {
+			$modules = Sensei()->modules->get_course_modules( $post_ID );
+			foreach ( $modules as $module ) {
+				self::update_module_teacher_meta( $module->term_id, $post_after->post_author );
+			}
+		}
+	}
+
+	/**
+	 * Add teacher id as term meta when a module is added to a course.
+	 *
+	 * @since $$next-version$$
+	 * @access private
+	 *
+	 * @param int    $object_id Object ID.
+	 * @param int    $tt_id     Term taxonomy ID.
+	 * @param string $taxonomy  Taxonomy slug.
+	 */
+	public function add_teacher_id_in_module_meta_when_added_to_course( int $object_id, int $tt_id, string $taxonomy ) {
+		if ( 'module' !== $taxonomy ) {
+			return;
+		}
+
+		$course = get_post( $object_id );
+
+		self::update_module_teacher_meta( $tt_id, $course->post_author );
+	}
+
+	/**
+	 * Remove teacher id from term meta when a module is added to a course.
+	 *
+	 * @since $$next-version$$
+	 * @access private
+	 *
+	 * @param int    $object_id Object ID.
+	 * @param array  $tt_ids    An array of term taxonomy IDs.
+	 * @param string $taxonomy  Taxonomy slug.
+	 */
+	public function remove_teacher_id_from_module_meta_when_removed_from_course( int $object_id, array $tt_ids, string $taxonomy ) {
+		if ( 'module' !== $taxonomy ) {
+			return;
+		}
+
+		foreach ( $tt_ids as $tt_id ) {
```

Comment:
What do you think about renaming these two variables to something longer but more meaningful? :)

Reply:
Hi @merkushin 👋 $tt_id is the documented name of the parameter for this term (like [here](https://developer.wordpress.org/reference/hooks/term_id_filter/)), so I thought it'd be better to keep it as it is